### ISSUE TITLE
Fix memory leak issue caught by dynamic analysis

### ISSUE THREAD

**Issue by ihhub** [ihhub/penguinV#486]
[Here](https://travis-ci.org/ihhub/penguinV/jobs/528804404) is a build in Travis-CI where we start having memory leak caught by dynamic analyzer. The issue was introduced by [this](https://github.com/ihhub/penguinV/commit/a0ed20d5e64b4fc2bc49a2403f377e0be50078d0) commit.

**Comment by ihhub** [comment]
Hi @yannlabou have you seen such problem previously? Looks like virtual machines on Travis-CI download wrong version of valgrind.

**Comment by ihhub** [comment]
To prove that this issue was not introduced by the change we ran unit tests without any tests [here](https://travis-ci.org/ihhub/penguinV/jobs/542612030).

**Comment by yannlabou** [comment]
> Hi @yannlabou have you seen such problem previously? Looks like virtual machines on Travis-CI download wrong version of valgrind.

I don't think that Travis-CI downloaded the wrong package version. `valgrind (1:3.10.1-1ubuntu3~14.5)` is the one available in Ubuntu Trusty 14.04 and `valgring (1:3.11.0-1ubuntu4.2)` is the package available in Ubuntu Xenial 16.04. According to the [Travis doc](https://docs.travis-ci.com/user/reference/overview/), the default linux distribution is `Trusty`. In the [build log](https://travis-ci.org/ihhub/penguinV/jobs/542612030#L9) you referenced, we are building on Xenial although we are not explicitly setting the linux distro in our  .travis.yml. Maybe Travis changed the default Ubuntu distro?

**Comment by ihhub** [comment]
I think they changed it.

**Comment by yannlabou** [comment]
> I think they changed it.

Looks like: https://blog.travis-ci.com/2019-04-15-xenial-default-build-environment

**Comment by ihhub** [comment]
We're going to try to run this job on GitHub Actions and hopefully we'll fix the problem.
